source-hubspot-native/source_hubspot_native/resources.py

```diff
@@ -22,56 +25,344 @@
+            inc=ResourceState.Incremental(cursor=started_at),
+            backfill=ResourceState.Backfill(next_page=None, cutoff=started_at),
+        ),
+        initial_config=ResourceConfig(name=cls.NAME, interval=timedelta(minutes=7)),
+        schema_inference=True,
+    )
+
+
+
+def crm_object_streamed_no_batch(
+    cls: type[V1CRMObject], http: HTTPSession, fetch_recent: FetchRecentFn
+) -> common.Resource:
+    """Custom Resource to run V1 API objects using stream
+    This resource does not use the batch function from "crm_object_streamed"
+    Allowing for a more broad usage.
+
+    Args:
+        cls (type[V1CRMObject]): _description_
+        http (HTTPSession): _description_
+        fetch_recent (FetchRecentFn): _description_
+
+    Returns:
+        common.Resource: _description_
+    """
+
+    def open(
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_changes=functools.partial(fetch_changes_no_batch, cls, fetch_recent, http),
+        )
+
+    started_at = datetime.now(tz=UTC)
+
+    return common.Resource(
+        name=cls.NAME,
+        key=cls.PRIMARY_KEY,
+        model=cls,
+        open=open,
+        initial_state=ResourceState(
+            inc=ResourceState.Incremental(cursor=started_at),
+            backfill=ResourceState.Backfill(next_page=None, cutoff=started_at),
+        ),
+        initial_config=ResourceConfig(name=cls.NAME, interval=timedelta(minutes=7)),
+        schema_inference=True,
+    )
+
+def workflow_object(
+    cls: type[V1CRMObject], http: HTTPSession, fetch_recent: FetchRecentFn
+) -> common.Resource:
+
+    """Custom Resource to run specifically workflow stream objects
+    Workflow objects do not have pagination neither a usual schema, 
+    so this resource was created to handle this specific case.
+    """
+    def open(
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_page=functools.partial(fetch_page_workflow, cls, http),
+        )
+
+    started_at = datetime.now(tz=UTC)
+
+    return common.Resource(
+        name=cls.NAME,
+        key=["/id"],
+        model=cls,
+        open=open,
+        initial_state=ResourceState(
+            inc=ResourceState.Incremental(cursor=started_at),
+            backfill=ResourceState.Backfill(next_page=None, cutoff=started_at),
+        ),
+        initial_config=ResourceConfig(name=cls.NAME, interval=timedelta(minutes=7)),
+        schema_inference=True,
+    )
+
+def subscription_object(
+    cls: type[V1CRMObject], http: HTTPSession, fetch_recent: FetchRecentFn
+) -> common.Resource:
+
```

Comment:
The comment for `workflow_object` is nice - as I'm reading this I am wondering why `subscription_object` exists, so a similar comment explaining that would be appreciated.

Reply:
A comment for this specific function was added here a484280098b57da301ffc5bc09ab89fa835fc8d0